drupal/sites/default/boinc/themes/boinc/templates/node.tpl.php

```diff
@@ -74,11 +74,14 @@
  */
 ?>
 
-<?php if ($subscribe_link): ?>
-  <div class="subscribe">
+<div class="subscribe">
+  <?php if (user_access('post comments')): ?>
+    <a href="#block-comment_form_block-comment_form">Post new comment</a>
+  <?php endif; ?>
+  <?php if ($subscribe_link): ?>
```

Comment:
This does not account for the case of logged in users that do not have permission to post comments, either due to ban or insufficient credit. In that case, the subscribe link will appear with a divider on the left side even though it is the first and only link. This is why links elsewhere are contained by a `<ul class="links">` structure with `<li>` elements that are given "first" or "last" class, as appropriate. The logic has to be a bit complicated, otherwise the styling breaks down. 

Related, it looks like the same or a similar bug exists in the comment template, which is rendering a divider to the right of the last link for users who are not allowed to post.

Reply:
The new commit changes the if statements around and uses the <ul> and <li> tags (and CSS classes) to generate the border between links.

The bug regarding comment links you found will be fixed separately.